src/include/OSL/rs_free_function.h

```diff
@@ -316,6 +316,13 @@ OSL_RSOP OSL_HOSTDEVICE bool
 rs_trace_get(OSL::OpaqueExecContextPtr oec, OSL::ustringhash name,
              OSL::TypeDesc type, void* val, bool derivatives);
 
+/// Allocates memory for a closure color. May return null if no memory could
+/// be allocated. It is the renderers responsibility to clean up these
+/// allocations after a shader is run and the closures have been processed.
```

Comment:
Please mention that this can be called from multiple threads and needs to be thread safe, but same applies to all free functions so add to top.